admin/registration_providers/views.py

```diff
@@ -356,3 +358,40 @@ def post(self, request, *args, **kwargs):
             }
         # Return a JsonResponse with the JSON error or the validation error if it's not doing an actual migration
         return JsonResponse(response_data)
+
+
+class ShareSourceRegistrationProvider(PermissionRequiredMixin, View):
+    permission_required = 'osf.change_registrationprovider'
+    raise_exception = True
+
+    def get(self, request, *args, **kwargs):
+        provider = RegistrationProvider.objects.get(id=self.kwargs['registration_provider_id'])
+
+        resp = requests.post(
+            f'{website_settings.SHARE_URL}api/v2/sources/',
+            json={
+                'data': {
+                    'type': 'Source',
+                    'attributes': {
+                        'homePage': provider.domain if provider.domain else f'{website_settings.DOMAIN}/registration/{provider._id}/',
+                        'longTitle': provider.name,
+                        'iconUrl': provider.get_asset_url('square_color_no_transparent')
+                    }
+                }
+            },
+            headers={
+                'Authorization': f'Bearer {website_settings.SHARE_API_TOKEN}',
+                'Content-Type': 'application/vnd.api+json'
+            }
+        )
+        resp.raise_for_status()
+        resp_json = resp.json()
+
+        provider.share_source = resp_json['data']['attributes']['longTitle']
+        for data in resp_json['included']:
+            if data['type'] == 'ShareUser':
+                provider.access_token = data['attributes']['token']
+
+        provider.save()
+
+        return redirect(reverse_lazy('registration_providers:share_source', kwargs={'registration_provider_id': provider.id}))
```

Comment:
the guts of this (making the request to share) duplicates [`ShareSourcePreprintProvider`](https://github.com/CenterForOpenScience/osf.io/blob/b80f2f50bb54a8c5b5259effe4d5437a74e021ef/admin/preprint_providers/views.py#L391-L435), which'll cost us in maintenance -- better to extract it to a shared function, or similar

also these safety checks from `ShareSourcePreprintProvider` (or something like them) might be nice to preserve in that shared place:
```py
        if preprint_provider.share_source or preprint_provider.access_token:
            raise ValueError('Cannot update share_source or access_token because one or the other already exists')
        if not osf_settings.SHARE_API_TOKEN or not osf_settings.SHARE_URL:
            raise ValueError('SHARE_API_TOKEN or SHARE_URL not set')
        if not preprint_provider.get_asset_url('square_color_no_transparent'):
            raise ValueError('Unable to find "square_color_no_transparent" icon for provider')
```